docs/source/configuration/header-propagation.mdx

```diff
@@ -6,4 +6,110 @@ description: Configuring the HTTP headers which are sent to subgraphs
+You can customize which HTTP headers the Apollo Router includes in its requests to your individual subgraphs. You specify these headers on a per-subgraph basis in your [YAML configuration file](#configuration-file).
+
+More specifically, each the headers plugin include can be configured with an array of header-related actions for the router to perform (in order) before sending a request:
+
+```yaml{4-12}
+plugins:
+  # ...other configuration...
+
+  apollo.headers:
+    all:
+    - propagate:
+        matching:
+          regex: ^upstream-header-.*
+    - remove:
+        name: "x-legacy-account-id"
+    subgraphs:
+      products:
+        - insert:
+            name: "router-subgraph-name"
+            value: "products"
+
+```
+
+The following transformations are specific to customizing headers sent to subgraphs:
+
+### `propagate`
+
+Enables you to selectively propagate headers that were included in the client's request to the router.
+
+> **Note:** The Router _never_ propagates so-called [hop-by-hop headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers#hop-by-hop_headers), such as `Content-Length`.
+
+You can specify which headers to propagate based on a matching [regex pattern](https://docs.rs/regex/latest/regex/):
+
+```yaml
+- propagate:
+    matching:
+      regex: .*
+```
+
+You can also provide a static string with the `named` option. These `named` configurations have additional flexibility, because they support the following options:
+
+- `default_value`: A value to set if _no_ value was sent by the client
+- `rename`: Renames the header's key to the provided value
+
+```yaml
+- propagate:
+      named: "x-user-id"
+      value: "abc123"
+      rename: "account-id"
+```
+
+### `remove`
+
+Enables you to selectively remove headers that were included in the client's request to the router. Similar to [`headers_propagate`](#headers-propagate), this option can match either a static string or a [regular expression](https://docs.rs/regex/latest/regex/).
+
+> **Note:** The Router _never_ propagates so-called [hop-by-hop headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers#hop-by-hop_headers), such as `Content-Length`.
+
+```yaml
+# Do not send this subgraph the "Cookie" header.
+- remove:
+    named: "Cookie"
+- remove:
+    matching:
+      # Remove headers that include the legacy 'x-' prefix.
+      regex: ^x-.*$
+```
+
+### `headers_insert`
+
+Enables you to add custom headers to requests going to a specific subgraph. These headers are always static strings that originate in the router, instead of originating in the client.
+
+```yaml
+- insert:
+    name: "sent-from-our-apollo-router"
+    value: "indeed"
+```
+
+### Example
+
+Here's an example of customizing different headers for two subgraphs: `accounts` and `products`. These two subgraphs are both sent `router-subgraph-name` headers with either the `accounts` or `products` value, respectively. Any other subgraph receives _no_ headers.
```

Comment:
This clarifying sentence doesn't reflect the configuration below, I don't think? Should reframe to match